src/pydap/tests/test_cas_esgf.py

```diff
@@ -0,0 +1,71 @@
+from pydap.client import open_url
+import pydap.net
+from pydap.cas import esgf
+import requests
+import numpy as np
+import os
+import logging
+
+import sys
+if sys.version_info < (2, 7):
+    import unittest2 as unittest
+else:
+    import unittest
+
+logging.basicConfig(filename='cas_esgf_test.log', level=logging.DEBUG)
+
+
+class TestESGF(unittest.TestCase):
+    url = ('http://cordexesg.dmi.dk/thredds/dodsC/cordex_general/'
+           'cordex/output/EUR-11/DMI/ICHEC-EC-EARTH/historical/r3i1p1/'
+           'DMI-HIRHAM5/v1/day/pr/v20131119/'
+           'pr_EUR-11_ICHEC-EC-EARTH_historical_r3i1p1_'
+           'DMI-HIRHAM5_v1_day_19960101-20001231.nc')
+
+    def test_basic_esgf_auth(self):
+        """Set up PyDAP to use the URS request() function"""
+        session = esgf.setup_session(os.environ['OPENID_ESGF'],
```

Comment:
I agree with you in principle w.r.t. ensuring that both URS and ESGF portals are accessible with pydap, but the challenges here are a real pain. We *could* use our personal credentials, but Travis doesn't make environment variables accessible to untrusted builds (e.g. pull requests). So basically, if we enable this, then *every other* pull request on this repo will fail the Travis build from here on out, regardless as to whether has anything to do with auth. I think that that's too big of an impact on the rest of the collaboration and build process. A few more comments below...